### ISSUE TITLE
phpstan errors after upgrading to 2.6.3 and doctrine/orm 2.12.2

### ISSUE THREAD

**Issue by usu** [doctrine/DoctrineBundle#1521]
After upgrading to DoctrineBundle 2.6.3 and doctrine/orm 2.12.2 I'm getting the following phpstan errors:

```
Parameter $repositoryClass of attribute class Doctrine\ORM\Mapping\Entity constructor expects class-string<Doctrine\ORM\EntityRepository<T of object>>|null, 'App\\Repository\\MyEntityRepository' given.
```

Is this an issue in DoctrineBundle? Or in doctrine/orm? Or an issue on my side?
I'm using phpstan 1.6.7 on level 5.

```php
namespace App\Entity;

use App\Repository\MyEntityRepository;

#[ORM\Entity(repositoryClass: MyEntityRepository::class)]
class MyEntity {
}
```

```php
namespace App\Repository;

use App\Entity\MyEntity;
use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
use Doctrine\Persistence\ManagerRegistry;

class MyEntityRepository extends ServiceEntityRepository {
    public function __construct(ManagerRegistry $registry) {
        parent::__construct($registry, MyEntity::class);
    }
}
```

**Comment by nicolas-grekas** [comment]
That issue is for maker-bundle, which should generate the annotation.
Not a bug here since that's "just" about a missing annotation.
`@extends ServiceEntityRepository<MyEntity>` I guess

**Comment by nicolas-grekas** [comment]
In theory, phpstan could be able to infer the type of the template by looking at the second argument of ServiceEntityRepository's call to parent::__construct(). Dunno how hard this would be to implement. That might be another issue for phpstan/psalm.

**Comment by usu** [comment]
Thanks for the quick feedback. I believe this was fixed in the meantime in phpstan (https://github.com/phpstan/phpstan/issues/7171) and shipped with phpstan 1.6.8 (https://github.com/phpstan/phpstan/releases/1.6.8)
